src/domain/flow-system/src/entities/flow/flow_outcome.rs

```diff
@@ -46,14 +56,17 @@ impl From<&ts::TaskResult> for FlowResult {
                 PullResult::UpToDate => Self::DatasetUpdate(FlowResultDatasetUpdate {
                     num_blocks: 0,
                     num_records: 0,
+                    watermark_modified: false,
                 }),
                 PullResult::Updated {
                     num_blocks,
                     num_records,
+                    new_watermark,
                     ..
                 } => Self::DatasetUpdate(FlowResultDatasetUpdate {
                     num_blocks,
                     num_records,
+                    watermark_modified: new_watermark.is_some(),
```

Comment:
Don't think this will give results you expect (see my other comment about semantics of `new_watermark`).